### DISCUSSION TITLE
posts/overuse-vim-visual-mode

### DISCUSSION THREAD

**Discussion by giscus** [m4xshen/m4xshen.dev#11] [Announcements]
# posts/overuse-vim-visual-mode

Learn why you might be overusing Vim's visual mode and how adopting a different mindset can help you use Vim more efficiently. Discover alternative commands and techniques that can save you keystrokes and boost your productivity in this insightful post.

https://m4xshen.dev/posts/overuse-vim-visual-mode

**Comment by hezirel** [comment]
Concerning copying, you can just do

```
%y*
```

To copy the full buffer into your system clipboard

**Reply by m4xshen** [reply]
Thanks for the great tips!

**Comment by not-leader** [comment]
hello, are you familiar with kakoune? it solves this problem by just making every action motion+operator

Example 1: `%<a-|>wl-copy<ret>` selecting all text with `%` and piping it to `wl-copy` because kakoune doesn't have a system clipboard register, though usually there's a keybind set up for the `wl-copy` pipe (in my config it would look like `%<spc>y`)

Example 2: `Kxd` using shift+k (`K`) to select previous line, `x` to widen selection to contain entire line, `d` to delete

and for bonus tip, `5Jxd`, similar to example 2

**Reply by not-leader** [reply]
okay the bonus tip seems to be `6Jxd` to mimic the vim action
